src/main/java/org/zone/region/flag/player/falldamage/PlayerFallDamageListener.java

```diff
@@ -0,0 +1,43 @@
+package org.zone.region.flag.player.falldamage;
+
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.block.CollideBlockEvent;
+import org.spongepowered.api.event.cause.entity.damage.source.DamageSource;
+import org.spongepowered.api.event.cause.entity.damage.source.DamageSources;
+import org.zone.ZonePlugin;
+import org.zone.region.Zone;
+import org.zone.region.group.Group;
+import org.zone.region.group.key.GroupKeys;
+
+import java.util.Optional;
+
+public class PlayerFallDamageListener {
+    @Listener
+    public void onPlayerFallDamageEvent(CollideBlockEvent.Fall event) {
+
+        if (!(event instanceof Player player)) {
+            return;
+        }
+
+        if (event.cause().first(DamageSource.class).isPresent()) {
```

Comment:
If the cause isnt present then it should be returned too as the only case it wont have a damage source is either a forge mod or a sponge plugin trying to emulate the event and failing

Reply:
.